packages/components/src/assay/AssayGridPanel.tsx

```diff
@@ -0,0 +1,463 @@
+                    </RequiresPermission>
+                    <RequiresPermission perms={PermissionTypes.Delete}>
+                        <MenuItem divider />
+                        <SelectionMenuItem
+                            id="assay-item-delete-menu-item"
+                            text="Delete"
+                            onClick={onDelete}
+                            queryModel={model}
+                            nounPlural={noun.toLowerCase()}
+                        />
+                    </RequiresPermission>
+                </ManageDropdownButton>
+            )}
+            {showSampleBtn && (
+                <SampleActionsButton model={model} user={user} metricFeatureArea="assayResultsSampleButton">
+                    {isWorkflowEnabled() && (
+                        <>
+                            <MenuItem header>Jobs</MenuItem>
+                            <JobsMenuOptionsComponent user={user} model={model} isAssay />
+                        </>
+                    )}
+                </SampleActionsButton>
+            )}
+        </div>
+    );
+});
+
+interface AssayGridPanelProps {
+    assayDefinition: AssayDefinitionModel;
+    canDelete?: boolean;
+    canUpdate?: boolean;
+    filters?: Filter.IFilter[];
+    getIsDirty?: () => boolean;
+    header?: string;
+    nounPlural?: string;
+    nounSingular?: string;
+    onEditToggle?: (isEditing: boolean) => any;
+    protocol?: AssayProtocolModel;
+    queryName: string;
+    requiredColumns?: string[];
+    setIsDirty?: (isDirty: boolean) => void;
+    showImport?: boolean;
+}
+
+export const AssayGridBodyImpl: FC<AssayGridPanelProps & InjectedQueryModels> = memo(props => {
+    const {
+        actions,
+        assayDefinition,
+        canDelete,
+        canUpdate,
+        header,
+        onEditToggle,
+        requiredColumns,
+        getIsDirty,
+        setIsDirty,
+        showImport = true,
+        queryModels,
+        nounSingular,
+        nounPlural,
+        protocol,
+        queryName,
+    } = props;
+
+    const [showConfirmDelete, setShowConfirmDelete] = useState<boolean>(false);
+    const [isEditing, setIsEditing] = useState<boolean>(false);
+    const [showBulkUpdate, setShowBulkUpdate] = useState<boolean>(false);
+    const [bulkFormUpdates, setBulkFormUpdates] = useState<OrderedMap<string, any>>(undefined);
+    const [selectionData, setSelectionData] = useState<Map<string, any>>(undefined);
+
+    const assayModel = useMemo<QueryModel>(() => Object.values(queryModels)[0], [queryModels]);
+    const modelId = useMemo(() => assayModel.id, [assayModel]);
+    const { createNotification, dismissNotifications } = useNotificationsContext();
+
+    const hideConfirm = useCallback(() => {
+        setShowConfirmDelete(false);
+    }, []);
+
+    const afterDelete = useCallback(() => {
+        hideConfirm();
+        onAssayRunChange(assayDefinition.protocolSchemaName);
+        actions.loadModel(modelId);
```

Comment:
LKB also had a call to `actions.clearSelections(queryModel.id);` is that needed here?

Reply:
Good catch. Thanks.